sdv/single_table/copulas.py

```diff
@@ -178,13 +178,21 @@ def get_learned_distributions(self):
         columns = parameters['columns']
         univariates = deepcopy(parameters['univariates'])
         learned_distributions = {}
+        metadata_columns = list(self.metadata._columns)
         for column, learned_params in zip(columns, univariates):
-            distribution = self.numerical_distributions.get(column, self.default_distribution)
-            learned_params.pop('type')
-            learned_distributions[column] = {
-                'distribution': distribution,
-                'learned_parameters': learned_params
-            }
+            for valid_column in metadata_columns:
```

Comment:
As opposed to looping through these each time, can we do one pass and just get all the valid columns and the just loop through those to set the distributions?